server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala

```diff
@@ -245,11 +254,22 @@ class DeltaSharedTable(
         filteredFilters.map { addFile =>
           val cloudPath = absolutePath(deltaLog.dataPath, addFile.path)
           val signedUrl = fileSigner.sign(cloudPath)
-          val modelAddFile = model.AddFile(url = signedUrl,
-            id = Hashing.md5().hashString(addFile.path, UTF_8).toString,
-            partitionValues = addFile.partitionValues,
-            size = addFile.size,
-            stats = addFile.stats)
+          val modelAddFile = if (Seq(version, timestamp).filter(_.isDefined).isEmpty) {
```

Comment:
same comment as before (pull this out of the loop) ?

Reply:
updated.